tasks/xpack/security/elasticsearch-security-native.yml

```diff
@@ -16,12 +16,30 @@
   set_fact: manage_native_roles=true
   when: es_roles is defined and es_roles.native is defined and es_roles.native.keys() | list | length > 0
 
+# For versions 7.x and above we need to use the /_security/... endpoint,
+# for versions 6.x and below we use the /_xpack/security/... endpoint
+- name: Only use _xpack in the security API calls for 6.x and below
+  set_fact:
+    es_security_endpoint: "{{'_security' if es_version is version_compare('7.0', operator='>=') else '_xpack/security'}}"
+
 #If the node has just has security installed it maybe either stopped or started 1. if stopped, we need to start to load native realms 2. if started, we need to restart to load
 
 #List current users
 - name: List Native Users
   uri:
-    url: http://{{es_api_host}}:{{es_api_port}}/_xpack/security/user
+    url: "http://{{es_api_host}}:{{es_api_port}}/{{es_security_endpoint}}/user"
```

Comment:
I'm would need to confirm this locally, but I believe that this being hardcoded to http means it will fail on the second run. We don't have the [idempotency_test](https://github.com/elastic/ansible-elasticsearch/blob/5b1d028bd2f81e343f7dc90c4f5b8b23ad8e80e5/.kitchen.yml#L130) enabled for this specific suite because it is supposed to test an upgrade path. So the tests pass because it first runs with http, then with https, though I fear it wil fail on the third fun when Elasticsearch is listening on https at this point